baselines/hfedxgboost/hfedxgboost/dataset_preparation.py

```diff
@@ -0,0 +1,219 @@
+
+    if dataset_name=="space_ga":
+        DATASET_PATH=os.path.join(ALL_DATASETS_PATH, "space_ga")
+        if not os.path.exists(DATASET_PATH):
+            os.makedirs(DATASET_PATH)
+            urllib.request.urlretrieve(
+                "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/space_ga",
+                f"{os.path.join(DATASET_PATH, 'space_ga')}",
+            )
+            urllib.request.urlretrieve(
+                "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/space_ga_scale",
+                f"{os.path.join(DATASET_PATH, 'space_ga_scale')}",
+            )
+        return [os.path.join(DATASET_PATH, 'space_ga_scale')]
+    if dataset_name=="abalone":
+        DATASET_PATH=os.path.join(ALL_DATASETS_PATH, "abalone")
+        if not os.path.exists(DATASET_PATH): 
+            os.makedirs(DATASET_PATH)       
+            urllib.request.urlretrieve(
+                "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/abalone",
+                f"{os.path.join(DATASET_PATH, 'abalone')}",
+            )
+            urllib.request.urlretrieve(
+                "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/abalone_scale",
+                f"{os.path.join(DATASET_PATH, 'abalone_scale')}",
+            )
+        return [os.path.join(DATASET_PATH, 'abalone_scale')]
+    if dataset_name=="cpusmall":
+        DATASET_PATH=os.path.join(ALL_DATASETS_PATH, "cpusmall")
+        if not os.path.exists(DATASET_PATH):  
+            os.makedirs(DATASET_PATH)      
+            urllib.request.urlretrieve(
+                "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/cpusmall",
+                f"{os.path.join(DATASET_PATH, 'cpusmall')}",
+            )
+            urllib.request.urlretrieve(
+                "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/cpusmall_scale",
+                f"{os.path.join(DATASET_PATH, 'cpusmall_scale')}",
+            )
+        return [os.path.join(DATASET_PATH, 'cpusmall_scale')]
+    
+def datafiles_fusion(data_paths):
+    """
+    Merge (if necessary) the data files and returns the features and labels sperated in
+    numpy arrays.
+    Args:
+        data_paths: List[Dataset Pathes]
+            - The pathes for the data that will be used in train and test, with train of full 
+            dataset in index 0    
+    Returns:
+        X: Numpy array
+            - The full features of the dataset.
+        y: Numpy array
+            - The full labels of the dataset.
+    """
+    data=load_svmlight_file(data_paths[0], zero_based=False)
+    X=data[0].toarray()
+    Y=data[1]
+    for i in range(1,len(data_paths)):
+        data= load_svmlight_file(data_paths[i], zero_based=False,n_features=X.shape[1])
+        X=np.concatenate((X,data[0].toarray()), axis=0)
+        Y=np.concatenate((Y,data[1]), axis=0)
+    return X,Y
+
+def train_test_split(X,y,train_ratio=.75):
```

Comment:
I wonder if this function should be changed so to randomly split the dataset into train:test (while still keeping the 0.75:0.25 train:test ratios as in the paper). The current code always takes the first 75% as training set...